packages/3d-web-client-core/src/character/CharacterModel.ts

```diff
@@ -244,13 +269,38 @@ export class CharacterModel {
     if (!targetAction) return;
 
     if (currentAction) {
-      currentAction.enabled = true;
       currentAction.fadeOut(transitionDuration);
     }
 
-    if (!targetAction.isRunning()) targetAction.play();
+    targetAction.reset();
+    if (!targetAction.isRunning()) {
+      targetAction.play();
+    }
+
+    if (targetAnimation === AnimationState.doubleJump) {
+      if (!this.preventDoubleJump) {
+        targetAction.setLoop(LoopRepeat, 1);
+        targetAction.clampWhenFinished = true;
+        targetAction.getMixer().addEventListener("finished", (_event) => {
+          if (this.currentAnimation === AnimationState.doubleJump) {
+            Object.values(this.animations).forEach((action) => {
+              action.stop();
+            });
+            this.preventDoubleJump = true;
```

Comment:
It took me a while to figure out what this was doing. If I understand correctly it's stopping the handling of a change back to `AnimationState.doubleJump` after the animation has played so that even though the controller is declaring the animation state is `AnimationState.doubleJump` the air animation isn't overridden. This was confusing to say the least 😅